docs/proposals/crd-implementation.md

```diff
@@ -0,0 +1,50 @@
+# Service Catalog migration to CRDs
+
+The Service Catalog migration to CRDs was done in [this](https://github.com/kyma-project/kyma/issues/3034) epic. You can find there all of the tasks which were necessary to complete this challenge.
+
+## Implemented features
+
+The API server custom features were migrated to a admission webhook approach.
+
+The table converter API server feature was migrated to the [Additional Printer Columns](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/#additional-printer-columns) feature.
+
+## Project changes
+
+The architecture after migration looks as follows:
+
+<img src="/docs/images/current.png" width="75%" height="75%">
+
+### Chart changes
+
+The following files were added to the Service Catalog helm chart:
+
+- **crds.yaml** which specifies the CustomResourceDefinitions for the ServiceCatalog resources. The CustomResourceDefinitions are installed via helm hook:
+```
+"helm.sh/hook": crd-install
+```
+- **webhook-deployment.yaml** - webhook server deployment.
+- **webhook-register.yaml** - registration of the webhook server operations.
+- **webhook-service.yaml** - service for the webhook server.
+
+These files were deleted from the Service Catalog helm chart because they are not necessary anymore due to the API server removal:
+
+- **apiregistration.yaml**
+- **apiserver-deployment.yaml**
+- **apiserver-service.yaml**
+- **etcd-pvc.yaml**
+
+All the webhook server configuration can be found under the `webhook` field in the `values.yaml`.
+
+### Code changes
+
+The following code changes were introduced to the Service Catalog:
+
+- **pkg/webhookutil** - under this directory you can find the common logic used in the webhook server implementation.
+- **pkg/webhook/servicecatalog** - here you can find the webhook server logic for all of the Service Catalog resources.
+- **cmd/webhook/server** - the implementation of the webhook server, here you can find the validation and mutation webhook registration.
+
```

Comment:
After merge PR #21 there will be extra:
- **pkg/cleaner** - responsible for removing CRD and CR during removing helm release
- **pkg/probe** - responsible for checking if webhook/controller is ready to use (readiness probe), in details if all required CRDs exist and have specific state